packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -275,10 +275,12 @@ export const useField = <
     event.preventDefault();
     onClear?.(event, ...(args as []));
     clearValue();
-    setSelectedSections(sectionOrder.startIndex);
 
-    if (!interactions.isFieldFocused) {
+    if (!interactions.isFieldFocused()) {
```

Comment:
:facepalm: :facepalm: :facepalm: :facepalm: :facepalm: :facepalm: :facepalm: :facepalm: 

Thanks for fixing it

Reply:
P.S. Didn't want to risk it again, that's why I took a bit longer than expected. 🙈 
At first, I tried tailoring a unit test—which proved to be nearly impossible to achieve reliably.
Then, I resorted to writing an e2e test, which also posed challenges due to Firefox's `getSelection` behavior... 🙈 🤷